warp10/src/main/java/io/warp10/continuum/gts/GTSEncoder.java

```diff
@@ -378,22 +386,26 @@ public synchronized int addValue(long timestamp, long location, long elevation,
       } else {
         tsTypeFlag |= FLAGS_BOOLEAN_VALUE_FALSE;
       }
-      
+
     } else if (value instanceof String) {
       tsTypeFlag |= FLAGS_TYPE_STRING;
-      if (((String) value).equals(lastStringValue)) {
+      if (validLastStringValue && ((String) value).equals(lastStringValue)) {
         tsTypeFlag |= FLAGS_VALUE_IDENTICAL;
       }
     } else if (value instanceof byte[]) {
       tsTypeFlag |= FLAGS_TYPE_STRING | FLAGS_STRING_BINARY;
       binaryString = new String((byte[]) value, StandardCharsets.ISO_8859_1);
-      if (binaryString.equals(lastStringValue)) {
+      if (validLastStringValue && binaryString.equals(lastStringValue)) {
         tsTypeFlag |= FLAGS_VALUE_IDENTICAL;
       }
     } else if (value instanceof Double || value instanceof Float) {
       tsTypeFlag |= FLAGS_TYPE_DOUBLE;
       // Only compare to the previous double value if the last floating point value was NOT encoded as a BigDecimal
-      if (null == lastBDValue && lastDoubleValue == ((Number) value).doubleValue()) {
+      if (validLastDoubleValue
+          && null == lastBDValue
```

Comment:
I think `validLastDoubleValue` and `validLastBDValue` could also be used to tell if the last floating-point value is encoded as BigDecimal or doubleinstead of setting `lastBDValue` to `null`:
- `validLastDoubleValue && !validLastBDValue` -> last value encoded as double
- `!validLastDoubleValue && validLastBDValue` -> last value encoded as BigDecimal
- `!validLastDoubleValue && !validLastBDValue` -> unknown last value floating-point value
- `validLastDoubleValue && validLastBDValue` -> cannot happen

Thus `validLastDoubleValue && null == lastBDValue` could be replaced by `validLastDoubleValue`.